src/main/kubeconfig-manager.ts

```diff
@@ -42,7 +48,7 @@ export class KubeconfigManager {
    * Creates new "temporary" kubeconfig that point to the kubectl-proxy.
    * This way any user of the config does not need to know anything about the auth etc. details.
    */
-  protected async createProxyKubeconfig(): Promise<string> {
+  protected createProxyKubeconfig(): string {
```

Comment:
does not need to be async

Reply:
Why are we going towards calling sync fuctions, that can block the event loop. We should move these towards the async methods instead.

Reply:
yes... I uses sync since this method already uses sync methods. I investigated option to switch to use async, but it would require bigger refactoring, so ended up to add smaller fix here.

Reply:
AFAIK the only async call there is `fs.ensureDir` which I think is probably wrong since it's not awaited.

Reply:
Actually I can be wrong, how big effort would it be to switch to use async... will check it again.

Reply:
Okay, opened the can of worms and I think I've found all the occurrences. WDYT should I keep these changes or revert to use `sync` versions?

Reply:
I think we should keep the `async`, yes it is larger but I think it is safer to not mix async and sync like we were.